### PR TITLE
feat: add service for setting announce volume

### PR DESCRIPTION
**Pull request by eyr1n**
## PR Type

- New Feature

## Description

-  add service for setting announce volume

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write release notes

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

**Comment by eyr1n** [comment]
Service名として適当に `/volume` 、またfloat32でvolumeというパラメータを生やした .srv ファイルを作成しました。
どのような名前にするべきかわからなかったので、変更していただければと思います。

よろしくお願いいたします。

**Review by tkhmy — COMMENTED** [review]
I added some comment

File: src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py [hunk 1]
```diff
@@ -352,3 +359,7 @@ def announce_stop_reason(self, file):
         self._in_stop_status = True
         self.send_announce(file)
         self.set_timeout("stop_reason")
+
+    def set_volume(self, request, response):
+        self._pulse.volume_set_all_chans(self._sink, request.volume)
```
[the remark this answers is not in the document]
Comment (eyr1n) [reply src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py:364]:
```py
    def set_volume(self, request, response):
        try:
            self._pulse.volume_set_all_chans(self._sink, request.volume)
            response.status.code = ResponseStatus.SUCCESS
        except Exception:
            response.status.code = ResponseStatus.ERROR
        return response
```

Is it ok? I'm using ResponseStatus in tier4_external_api_msgs package.

[the remark this answers is not in the document]
Comment (tkhmy) [reply src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py:364]: that's look good for me

[the remark this answers is not in the document]
Comment (eyr1n) [reply src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py:364]: Thank you.

File: src/vehicle_voice_alert_system_interfaces/srv/Volume.srv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+float32 volume
+---
```
[the remark this answers is not in the document]
Comment (eyr1n) [reply src/vehicle_voice_alert_system_interfaces/srv/Volume.srv:2]: moved the message by this PR
https://github.com/tier4/tier4_autoware_msgs/pull/104

File: src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py [hunk 3]
```diff
@@ -85,6 +87,11 @@ def __init__(
         self._node.create_timer(0.5, self.stop_reason_checker_callback)
         self._node.create_timer(0.1, self.announce_engage_when_starting)
 
+        self._pulse = Pulse()
+        # Get default sink at startup
+        self._sink = self._pulse.get_sink_by_name(self._pulse.server_info().default_sink_name)
+        self._node.create_service(Volume, "/volume", self.set_volume)
```
[the remark this answers is not in the document]
Comment (eyr1n) [reply src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py]: fixed

File: README.md [hunk 4]
```diff
@@ -23,6 +23,12 @@ refer to here
 sudo pip3 install simpleaudio
 ```
 
+- pulsectl
+
+```bash
+sudo pip3 install pulsectl
```
[the remark this answers is not in the document]
Comment (eyr1n) [reply README.md]: fixed

**Comment by tkhmy** [comment]
@eyr1n Thanks it looks good for me.
I would like to test in the bench first before we merge it.
On the other hand, can you fix the per-commit error

File: src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py [hunk 5]
```diff
@@ -85,6 +88,11 @@ def __init__(
         self._node.create_timer(0.5, self.stop_reason_checker_callback)
         self._node.create_timer(0.1, self.announce_engage_when_starting)
 
+        self._pulse = Pulse()
+        # Get default sink at startup
+        self._sink = self._pulse.get_sink_by_name(self._pulse.server_info().default_sink_name)
+        self._node.create_service(Volume, "~/volume", self.set_volume)
```
Comment (tkhmy) [inline src/vehicle_voice_alert_system/src/vehicle_voice_alert_system/announce_controller.py:94]: the name change to ~/set_volume will be better for understanding

**Review by tkhmy — APPROVED** [review]
LGTM

**Pull request merged**